@xen-orchestra/lite/src/components/Tree.tsx

```diff
@@ -116,23 +129,66 @@ const Tree = withState<State, Props, Effects, Computed, ParentState, ParentEffec
   {
     initialState: ({ defaultSelectedNodes }) => ({
       selectedNodes: defaultSelectedNodes === undefined ? [] : defaultSelectedNodes,
+      _expandedNodes: undefined,
     }),
     effects: {
+      setExpandedNodeIds: function (event, nodeIds) {
+        this.state._expandedNodes = nodeIds
+      },
       setSelectedNodeIds: function (event, nodeIds) {
         this.state.selectedNodes = nodeIds
       },
     },
+    computed: {
+      expandedNodes: ({ _expandedNodes }, { collection, defaultSelectedNodes }) => {
+        if (defaultSelectedNodes === undefined) {
+          return _expandedNodes ?? [collection[0].id]
+        }
+
+        let expandedNodes: Array<string> = _expandedNodes ?? []
+        const tempExpandedNodes: Set<string> = new Set()
+
+        const addExpandedNode = (collection: Array<ItemType> | undefined) => {
+          if (collection === undefined) {
+            return
+          }
+
+          let isExpandedNode = false
+          for (const node of collection) {
+            const children = node.children
+            if (children === undefined) {
+              if (defaultSelectedNodes.includes(node.id)) {
+                expandedNodes = expandedNodes.concat(Array.from(tempExpandedNodes))
+                isExpandedNode = true
+              }
+            } else {
+              tempExpandedNodes.add(node.id)
+              addExpandedNode(node.children)
+              if (!isExpandedNode) {
+                tempExpandedNodes.delete(node.id)
+              }
```

Comment:
I think you shouldn't add and then delete node : 
```suggestion
              addExpandedNode(node.children)
              if (isExpandedNode) {
                  tempExpandedNodes.add(node.id)
              }
```

Reply:
Now, I delete the node after building `expandedNodes` above (recursive method).